DragonSlayer.cpp

```diff
@@ -1,20 +1,25 @@
 #include "DragonSlayer.h"
 #include "Dragon.h"
+#include "Utility.h"
 
-//DragonSlayer::DragonSlayer
-
-//DragonSlayer::getName
+DragonSlayer::DragonSlayer(const std::string& name_, int hp_, int armor_) : Character(hp_, armor_, 4), name(name_)
+{
+    helpfulItems = makeHelpfulItems(1);
+    defensiveItems = makeDefensiveItems(1);
+}
 
 void DragonSlayer::attack(Character& other)
 {
     std::cout << name << " is attacking " << other.getName() << " !!" << std::endl;
     if( auto* dragon = dynamic_cast<Dragon*>(&other) )
     {
-        assert(false);
         //DragonSlayers get a 10x boost when attacking dragons, from their attack item.
         //so they should USE their attack item before attacking the dragon... 
         //note that items are single-use only, so you need to reset it after use.  
         //look in the Character class for how the other item types are reset after use.
+
+        attackItem.use(this);
+
```

Comment:
You need to make sure your `attackItem` is valid before you try to use it.
(hint: check for nullptr)
Also, you need to reset this after it's used.
```
        //note that items are single-use only, so you need to reset it after use.  
        //look in the Character class for how the other item types are reset after use.
```
- [ ] revise